server/routes/utils/dataReportHelpers.ts

```diff
@@ -114,32 +119,67 @@ export const buildRequestBody = (report: any, allowLeadingWildcards: boolean, is
 
 // Fetch the data from OpenSearch
 export const getOpenSearchData = (
-  arrayHits,
-  report,
-  params,
+  arrayHits: any,
+  report: { _source: any },
+  params: { excel: any; limit: number },
   dateFormat: string
 ) => {
   let hits: any = [];
   for (let valueRes of arrayHits) {
     for (let data of valueRes.hits) {
       const fields = data.fields;
       // get all the fields of type date and format them to excel format
+      let tempKeyElement: string[] = [];
       for (let dateField of report._source.dateFields) {
+        let keys;
+        keys = dateField.split('.');
         const dateValue = data._source[dateField];
-        if (dateValue && dateValue.length !== 0) {
-          if (dateValue instanceof Array) {
-            // loop through array
-            dateValue.forEach((element, index) => {
-              data._source[dateField][index] = moment(
-                fields[dateField][index]
-              ).format(dateFormat);
+        const fieldDateValue = fields[dateField];
+        // if its not a nested date field
+        if (keys.length === 1) {
+          // if conditions to determine if the date field's value is an array or a string
+          if (typeof dateValue === 'string') {
+            data._source[keys] = moment(dateValue).format(dateFormat);
+          } else if (
+            fieldDateValue.length !== 0 &&
+            fieldDateValue instanceof Array
+          ) {
+            console.log('entered');
```

Comment:
minor: can remove the console log from here.

Reply:
done